### ISSUE TITLE
“Observable Projector not supported on device lightning.gpu”

### ISSUE THREAD

**Issue by Leisurivan** [PennyLaneAI/pennylane-lightning#883]
# I find “Observable Projector not supported on device lightning.gpu”, but it's said to be supported in the stable version document.

### When I run the following codes, there will be no error.
```
dev = qml.device("lightning.gpu", wires=num_wires)
wires = list(range(num_wires))
@qml.qnode(dev, diff_method="adjoint")
def kernel(x1, x2, params):
    ansatz(x1, params, wires)
    qml.adjoint(ansatz)(x2, params, wires)
    return qml.expval(qml.Projector([0]*num_wires, wires=wires))
```
### But when I try to get the result, it will output “Observable Projector not supported on device lightning.gpu”
```
result = kernel(X1, X2, params)
```
### Here is the device information:

Name: PennyLane
Version: 0.37.0
Summary: PennyLane is a cross-platform Python library for quantum computing, quantum machine learning, and quantum chemistry. Train a quantum computer the same way as a neural network.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /root/anaconda3/envs/pennylane_gpu/lib/python3.11/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, packaging, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml, typing-extensions
Required-by: PennyLane_Lightning, PennyLane_Lightning_GPU

Platform info:           Linux-5.4.0-144-generic-x86_64-with-glibc2.31
Python version:          3.11.9
Numpy version:           1.26.4
Scipy version:           1.13.1
Installed devices:
- lightning.qubit (PennyLane_Lightning-0.37.0)
- default.clifford (PennyLane-0.37.0)
- default.gaussian (PennyLane-0.37.0)
- default.mixed (PennyLane-0.37.0)
- default.qubit (PennyLane-0.37.0)
- default.qubit.autograd (PennyLane-0.37.0)
- default.qubit.jax (PennyLane-0.37.0)
- default.qubit.legacy (PennyLane-0.37.0)
- default.qubit.tf (PennyLane-0.37.0)
- default.qubit.torch (PennyLane-0.37.0)
- default.qutrit (PennyLane-0.37.0)
- default.qutrit.mixed (PennyLane-0.37.0)
- default.tensor (PennyLane-0.37.0)
- null.qubit (PennyLane-0.37.0)
- lightning.gpu (PennyLane_Lightning_GPU-0.37.0)

**Comment by maliasadi** [comment]
Hey @Leisurivan, thanks for reporting this issue.

> I find “Observable Projector not supported on device lightning.gpu”, but it's said to be supported in the stable version document.

This was indeed a literal error in our public documentations. This is fixed in v0.38.0 ([lightning_gpu/supported-operations-and-observables](https://docs.pennylane.ai/projects/lightning/en/stable/lightning_gpu/device.html#supported-operations-and-observables)).

> But when I try to get the result, it will output “Observable Projector not supported on device lightning.gpu”

We can provide the support via diagonalizing the operation. This wouldn't be as efficient as a native implementation in C++, but it should unblock you for now.

**Comment by Leisurivan** [comment]
> 嘿，感谢您报告此问题。
> 
> > 我发现“设备 lightning.gpu 不支持可观察投影仪”，但据说在稳定版文档中支持它。
> 
> 这确实是我们公共文档中的字面错误。此问题已在 v0.38.0 （[lightning_gpu/supported-operations-and-observables](https://docs.pennylane.ai/projects/lightning/en/stable/lightning_gpu/device.html#supported-operations-and-observables)） 中修复。
> 
> > 但是当我尝试获取结果时，它会输出“设备 lightning.gpu 不支持可观察投影仪”
> 
> 我们可以通过对角化操作来提供支持。这不如 C++ 中的本机实现有效，但现在应该可以解除阻塞。

Thanks！
